pkg/cmd/pulumi/new.go

```diff
@@ -1112,3 +1117,29 @@ func containsWhiteSpace(value string) bool {
 	}
 	return false
 }
+
+func validateStackProjectName(b backend.Backend, stackName, projectName string) error {
+	if stackName == "" {
+		return nil
+	}
+	if projectName == "" {
+		return nil
```

Comment:
optional nit: In this case, the two conditions are simple enough to combine, IMO.

```suggestion
	if stackName == "" || projectName == "" {
		return nil
```